src/lie_solvers.jl

```diff
@@ -0,0 +1,226 @@
+
+
+
+@doc raw"""
+    apply_diff_group(A::AbstractGroupAction, a, X, p)
+
+For a point on manifold ``p ∈ \mathcal M`` and an element `X` of the tangent space at `a`,
```

Comment:
I thought we had something like this already in Manifolds? I thought it would be https://juliamanifolds.github.io/Manifolds.jl/latest/manifolds/group.html#Manifolds.apply_diff-Tuple{AbstractGroupAction,%20Any,%20Any,%20Any} , since apply was anyways only for groups?

Reply:
The way to think about it is, `apply` takes group element `a`, point `p`, and gives point `q`. `apply_diff` is the differential with respect to `p` while `apply_diff_group` is the differential with respect to `a`. Do you have a better name for this?

Reply:
For this one here it would be enough to just include the image at the end I think; Pluto is something I want to try if I find time. Maybe first on Manopt.jl; there is a repo with examples how to render Pluto notebooks to html and include them in the docs.

Reply:
We'd probably want this function in Manifolds.jl but to get it working quickly I've put it here.

Reply:
Hm, why did my plot comment end up here? I was sure to have written it a comment field above...

However. Then I have not yet understood the difference of `apply_diff` and `apply_diff_group`, when `apply` is anyways only available on groups.

Reply:
Sometimes github acts weirdly about comments...

Anyway, regarding `apply_diff`, another way to think about it is that `apply_diff` is the pushforward for a fixed `a` while `apply_diff_group` is the pushforward for a fixed `p`.

Reply:
Ah ok, then we should move that over to Manifolds at some point, yes. Thanks for the clarification.